openvasd/openvasd.c

```diff
@@ -598,7 +615,12 @@ openvasd_get_vts_stream_init (openvasd_connector_t *conn,
 
   h = curl_multi_init ();
   curl_multi_add_handle (h, hnd);
-  *mhnd = h;
+
+  if (mhnd == NULL)
+    mhnd = openvasd_curlm_handler_new ();
```

Comment:
Pointers are passed by value, so you need to pass `openvasd_curlm_t **mhnd` and assign the new struct to `*mhnd`. I hold on to it like this:

Think of a pointer variable as an int. It holds a number that is the index into the process's memory. The caller has a variable with this "int". The caller sends you the value of this int. When you assign to your int variable, only the value of your int changes. Whatever the caller has in their int variable stays the same. And whatever was at that index in memory stays the same.

The pointer to a pointer is like having a piece of memory with the "int" stored in the memory. When you assign to `*mhnd`
the value in the memory changes, and the caller gets the change because their "int" is located at the same memory. (There are more details, like the caller's variable may be a register, or stack allocated, or whatever, which the compiler sorts out.) 

Might help to printf the mhnd with %p in the function and in the caller, to get a sense of it.

Reply:
Is the user always going to create both openvasd_connector_t * and openvasd_curlm_t * together? Then you put the openvasd_curlm_t inside the openvas_connector_t and just allocate it always yourself when the openvas_connector_t is created.

Reply:
> Is the user always going to create both openvasd_connector_t * and openvasd_curlm_t * together?
Hi @mattmundell ! Thanks a lot for the review. Regarding this question, not always an `openvasd_curlm_t` is used which uses the curl multiperform, allowing to fetch data chunk by chunk. This will be used only for fetching the feed metadata (~ 300MB).
Other requests with smaller responses, perform the request using another curl handler type (curl easy handler). This will give the whole response at once. 
In the first implementation, the user didn't have access to the handler (still not for some request using the easy handler). But it has access to the multiperform handler for fetching the feed. 
Therefore, the handler is not inside the connector.